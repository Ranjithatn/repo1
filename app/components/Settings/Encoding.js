import React from 'react';

const Encoding = (props) => {

  const { formatMessage, nfiq } = props;

  let imageFormat = 'Unspecified';
  let templateFormat = 'Unspecified';

  if ( Array.isArray(nfiq) ) {
    const imageFormatObj = nfiq.find( data => data.setting === "Biometrics.Format.DefaultImageEncoding" );
    if ( imageFormatObj && imageFormatObj.value ) { imageFormat = imageFormatObj.value; }
    const templateFormatObj = nfiq.find( data => data.setting === "Biometrics.Format.DefaultTemplateType" );
    if ( templateFormatObj && templateFormatObj.value ) { templateFormat = templateFormatObj.value; }
  }



  return (
    <div>
      <h1>{ formatMessage({ id: "SettingsPageEncoding" }) }</h1>

      <div className="input-wrapper">
        <label>{ formatMessage({ id: "SelectedImageFormat" }) }</label>
        <input disabled value={ imageFormat } className="input" />
      </div>

      <div className="input-wrapper">
        <label>{ formatMessage({ id: "SelectedTemplateFormat" }) }</label>
        <input disabled value={ templateFormat } className="input" />
      </div>

    </div>
  )

}


export default Encoding;
