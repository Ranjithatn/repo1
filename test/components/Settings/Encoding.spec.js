import React from "react";
import toJson from "enzyme-to-json";
import Encoding from "../../../app/components/Settings/Encoding";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";


configure({ adapter: new Adapter() });

const props = {
  formatMessage: data => data.id

}

describe('<Encoding />', () => {

  it("should render without crashing", () => {
    const component = mount(
      <Encoding {...props}/>
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

});
