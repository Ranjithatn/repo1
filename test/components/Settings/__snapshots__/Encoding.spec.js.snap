// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Encoding /> should render without crashing 1`] = `
<Encoding
  formatMessage={[Function]}
>
  <div>
    <h1>
      SettingsPageEncoding
    </h1>
    <div
      className="input-wrapper"
    >
      <label>
        SelectedImageFormat
      </label>
      <input
        className="input"
        disabled={true}
        value="Unspecified"
      />
    </div>
    <div
      className="input-wrapper"
    >
      <label>
        SelectedTemplateFormat
      </label>
      <input
        className="input"
        disabled={true}
        value="Unspecified"
      />
    </div>
  </div>
</Encoding>
`;
